### ISSUE TITLE
Adding elevation to primitive layer

### ISSUE THREAD

**Issue by shirazb** [uber/deck.gl#2191]
Hi,

Before I begin, I would like to take a moment to highly commend the efforts of the Uber vis team. The entire suite of frameworks are great and very well maintained.

Now, onto my problem.

I have created a primitive layer containing an instanced 3D luma.gl model. Without elevation, all is good. However, this (seemingly trivial) task is getting the better of me. Currently, the vertex shader is similar to that of the hexagon cell layer, along the lines of:

``` glsl
vec3 scaledLocalPos = project_scale(positions);
vec3 orientedLocalPos = rotate(scaledLocalPos);

float elevation = instanceElevations * ELEVATION_SCALE;
vec3 elevatedInstancePosition = vec3(instancePositions, elevation);

vec4 position_worldspace;
gl_Position = project_position_to_clipspace(
        elevatedInstancePosition,
        instancePositions64xyLow,
        orientedLocalPos,
        position_worldspace
);

// Set the varying that controls colour...
```

When doing this, no elevation occurs. I have done a little debugging:

- By setting `luma.log.priority` in the browser dev tools, I can see that instanceElevations is indeed being passed to the GPU correctly.
- I have multiple of these layers on the map. If I give only one of them elevation, and set the colour based on `elevatedInstancePosition.z`, there is a clear difference for that one layer.
- If I set the colour based on `position_worldspace.y` there is no difference.

This leads me to believe I am using the `project_position_to_clipspace` function incorrectly. Unfortunately, other than setting the colour I am not really sure how to debug shader code, so haven't been able to investigate further.

Any help on this would be much appreciated.

Thanks,
Shiraz

**Comment by Pessimistress** [comment]
Is `ELEVATION_SCALE` a uniform? Can you check if it's properly set?

**Comment by ibgreen** [comment]
To do that, type `luma.log.priority = 3` in your browser console and rerun your app, you should see a table of uniform and attribute values printed in the console, check that nothing is undefined and that the values look good.

**Comment by shirazb** [comment]
It is merely a `#define ELEVATION_SCALE 0.8`. I went ahead and double checked the rest of the attributes and uniforms though. They seem fine, except maybe `instancePositions64xyLow`.

To be honest, I have that in there from when I first was experimenting with custom layers a while ago, but I don't fully know how it works nor am I really using it (though I plan to in the future). Currently, I set the attribute to `new Float32Array(2)`. Or, taken verbatim from the luma log output:
``` 
instancePositions64xyLow          "[0.,0.]"           "I  1 (x2=8bytes undefined) loc=3"
```
